src/renderer/model/battle/abstract-battle.ts

```diff
@@ -90,6 +90,19 @@ export abstract class AbstractBattle<T extends BattleOptions = BattleOptions> {
         });
     }
 
+    // Returns a bot by it's PlayerId. Returns undefined if not found..
+    public getBotParticipantByPlayerId(playerId: number): User | Bot | undefined {
+        return this.participants.value.find((participant) => {
+            const isBot = !("userId" in participant);
+            if (isBot) {
+                if (participant.playerId === playerId) {
+                    return true;
+                }
+            }
+            return false;
+        });
+    }
+
```

Comment:
I think this should `.find` on `this.bots` instead of `this.participants`, then you can remove `User` from the return type

Reply:
Thanks for the suggestion!

Addressed in 7bcc76.